botocore/compat.py

```diff
@@ -360,3 +360,20 @@ def has_minimum_crt_version(minimum_version):
     HAS_GZIP = True
 except ImportError:
     HAS_GZIP = False
+
+
+def __getattr__(name):
+    """
+    Module override to raise warnings when deprecated libraries
+    are imported in external code.
+    """
+    if name == "six":
+        warnstr = (
+            "The botocore.compat.six module is deprecated and will be removed "
+            "in a future version. Please use six as a direct dependency."
+        )
+        warnings.warn(warnstr, DeprecationWarning, stacklevel=2)
+        from botocore.vendored import six
+        return six
+
+    raise AttributeError(f"Module {__name__} has no attribute {name}.")
```

Comment:
Why isn't this breaking?  Probably something I'm missing, but wouldn't this fail to return existing attributes on the module and always raise the error?

Reply:
> Why isn't this breaking? Probably something I'm missing, but wouldn't this fail to return existing attributes on the module and always raise the error on any attribute other than six?

This is the fallback to the standard import mechanism.

> Also since you removed the import statement above, wouldn't this just be breaking now?

This is the import statement now. Imports are going through `__getattr__`. You can try it out with the PR and our existing test suite.

Reply:
If you're not familiar with module level __getattr__, [PEP 562](https://peps.python.org/pep-0562/#rationale) is a good read. That's what this implementation is based off.